api/controllers/node.js

```diff
@@ -49,34 +60,39 @@ function NodeController(scope) {
  * @param {function} next
  * @todo Add description for the function and the params
  */
-NodeController.getConstants = function(context, next) {
-	return modules.node.shared.getConstants(null, (err, data) => {
-		try {
-			if (err) {
-				return next(err);
-			}
-
-			data = _.cloneDeep(data);
-
-			// Perform required typecasts for integer
-			// or bignum properties when returning an API response
-			data.supply = data.supply.toString();
-			data.milestone = data.milestone.toString();
-			data.reward = data.reward.toString();
-			data.fees.dappDeposit = data.fees.dappDeposit.toString();
-			data.fees.dappWithdrawal = data.fees.dappWithdrawal.toString();
-			data.fees.dappRegistration = data.fees.dappRegistration.toString();
-			data.fees.multisignature = data.fees.multisignature.toString();
-			data.fees.delegate = data.fees.delegate.toString();
-			data.fees.secondSignature = data.fees.secondSignature.toString();
-			data.fees.vote = data.fees.vote.toString();
-			data.fees.send = data.fees.send.toString();
-
-			return next(null, data);
-		} catch (error) {
-			return next(error);
-		}
-	});
+NodeController.getConstants = async (context, next) => {
+	try {
+		const [lastBlock] = await library.storage.entities.Block.get(
+			{},
+			{ sort: 'height:desc', limit: 1 }
+		);
+		const { height } = lastBlock;
+
+		return next(null, {
+			build: library.build,
+			commit: library.lastCommit,
+			epoch: EPOCH_TIME,
+			fees: {
+				send: FEES.SEND.toString(),
+				vote: FEES.VOTE.toString(),
+				secondSignature: FEES.SECOND_SIGNATURE.toString(),
+				delegate: FEES.DELEGATE.toString(),
+				multisignature: FEES.MULTISIGNATURE.toString(),
+				dappRegistration: FEES.DAPP_REGISTRATION.toString(),
+				dappWithdrawal: FEES.DAPP_WITHDRAWAL.toString(),
+				dappDeposit: FEES.DAPP_DEPOSIT.toString(),
+			},
+			nethash: library.config.nethash,
+			nonce: library.config.nonce,
+			milestone: blockReward.calcMilestone(height).toString(),
+			reward: blockReward.calcReward(height).toString(),
+			supply: blockReward.calcSupply(height).toString(),
+			version: library.config.version,
+			protocolVersion: library.config.protocolVersion,
```

Comment:
I suspect this attribute was not already there.  If you want to introduce it now, then must add to `swagger` specification as well.

Reply:
@nazarhussain it was introduced on LIP007 which is already merged on development branch. [here](https://github.com/LiskHQ/lisk/blob/2651b07d0023f8ff7665823bf1c810e2b56b2575/schema/swagger.yml#L2318)